### DISCUSSION TITLE
Configure babel-plugin-transform-react-jsx question

### DISCUSSION THREAD

**Discussion by dedalqq** [babel/babel#15506] [Q&A]
Hi all. I Use babel-plugin-transform-react-jsx without react and use my implementation of jsx function for create DOM element. And if i use `importSource` than jsx will pass children as a atributes options. If i use `pragma` children will be pass as third parameter.

esample:
```
// input
<div>text</div>

jsx("div", {children: "text"}) // for importSource

jsx("div", {}, "text") // for pragma
```

I want use importSource because it support auto import but also i want get children as third parameter. What do I need to do for this? =)))

I think that use children as third parameter is a bad idea because i get this incorrect result for custom element. Example:
```
// input
<MyElement children="bar">foo</MyElement>

// output
jsx(MyElement , {
children: "bar",
children: "foo",
})
```

thenks =)

**Comment by nicolo-ribaudo** [comment]
`babel-plugin-transform-react-jsx` follows the de-facto standard set by React. There are in practice two API shapes for JSX implementations:
- "classic", which relies on a variable already explicitly defined in the current scope (for example, `React.createElement`) and passes children as a third parameter
- "automatic", which doesn't require developers using JSX to manually import the apparently-unused JSX factory function, and that passes children as part of the second parameter

Unfortunately there is no middle ground, because it is not deemed necessary by React or the other React-like popular libraries.

However, this question makes me notice that Babel could improve here, by throwing/warning when `children` is used as an attribute with the automatic runtime.
